### ISSUE TITLE
Evenly Distributed Reach Lengths

### ISSUE THREAD

**Issue by GregCocks-NOAA** [NOAA-OWP/cahaba#193]
_Task(s) Request Based On Fernando Salas @ 01/05/20 email_

**BACKGROUND**:
- OWP working on leveraging NHDPlus-MR/HR reaches to support streamflow prediction calculations and inundation extent calculations
- FIM team focused on the latter as we use the streamflow forecasts from the National Water Model (NWM), developed by the Analysis and Prediction Division (APD), as input to our FIM prediction capabilities.
- Inland Hydraulics project is working on developing a more sophisticated hydraulic routing module to the NWM implementation team for NWM v3.0 (OWP is about to implement NWM v2.1 later in Jan. ‘21). In order to support a more sophisticated hydraulic routing module, the team needs a river network with reach lengths that are around 400 meters in length; 600 meters max in length and 250 meters min in length. The Inland Hydraulics team is currently using the NWM v2.1 channel network as the starting point which is primarily derived from the reaches in v2.1 of NHDPlus-MR.
-  FIM team has been working on transforming NHDPlus-HR reaches in such a way that the segmentation is more evenly distributed in length (around 1-2 kms), to support FIM prediction capabilities with better performing synthetic rating curves
- In parallel to our efforts, the NWM development team has also been working to achieve a similar result.
- Although the goals of our two teams are more or less one in the same, at this time, the hydrofabric of our FIM capabilities is different than the hydrofabric of the NWM capabilities and as such the two tasks are somewhat independent; this is something we want to align in the long run but not critical in the short term. Yes, both FIM and NWM use NHDPlus as a starting point but the underlying network we implement is different.

**NWM HYDROFABRIC REFERENCES:**
 

- the network information for NWM v2.1

1. geometry in GDBs
2. network info in routeLink netCDF files
3. <nwm_geospatial_v2_1_final_20200622.zip>
4. <nwm_routelink_v2_1_final_20200622.zip>

- GitHub repo for the Inland Hydraulics project where they have already made an attempt to implement a segmentation solution (let's call it their alpha solution) but would like to refine it over the next few months: https://github.com/NOAA-OWP/t-route 

**TASK(S) GOAL**:
1. **Primary**: Inland Hydraulic Team needs a river network with reach lengths that are around 400 meters in length; 600 meters max in length and 250 meters min in length.
2. _Secondary_: (potentially) a goal for the FIM team; this would be tracked in a separate ticket
3. _Tertiary_: over time try to bring the NWM and FIM efforts together so we have one consistent hydrofabric data model to support both efforts

**PROCESSES**:
- @GregCocks-NOAA to connect with Inland Hydraulics project on 01/11/21; (@jameshalgren is the technical lead) to help them achieve their goal of implementing a network with a more evenly distributed reach length
- @GregCocks-NOAA to  spend ½ time working with the Inland Hydraulics project through March unless we achieve a stable solution earlier
- if more time needed, coordinate with @BradfordBates-NOAA, @ChristyWestcott-NOAA  and the Inland Hydraulics project to determine how much more time is needed to achieve the desired result

**Comment by jameshalgren** [comment]
Tag@awlostowski-noaa

**Comment by GregCocks-NOAA** [comment]
The less refined notes from [ticket 179](https://github.com/NOAA-OWP/cahaba/issues/179) (now marked closed):

**Comment by GregCocks-NOAA** [comment]
Covered in other cards the immediate need for helping @awlostowski-noaa  and @jameshalgren with Inland Hydraulics short segments as part of their RouteLink α solution adjustments...
[ShortSegments_Vectors_BusinessRules_PseudoCode_working_Jan21_GregCocks.xlsx](https://github.com/NOAA-OWP/cahaba/files/5955154/ShortSegments_Vectors_BusinessRules_PseudoCode_working_Jan21_GregCocks.xlsx)

However, MIGHT return to this with and for @BrianAvant-NOAA as part of FIM work directly; the attached business rules and pseudocode for vectors working document might provide some guidance / approaches
